### PR TITLE
[python] added get_split_value_histogram method

### LINKED ISSUE #2013: get_split_value_histogram XGBoost equivalent in LightGBM
I want histogram/values of all the splits happening in a particular feature. Just like get_split_value_histogram in XGBoost. Is there any way to get it currently? 

I am aware of plot_tree and create_tree_digraph methods. But those are for visualizations

### PR DESCRIPTION
**Pull request by StrikerRUS**
Closed #2013.

Calculate feature split histogram in XGBoost-style (https://github.com/dmlc/xgboost/blob/master/python-package/xgboost/core.py#L1694-L1737).

**UPD:** … and in standard (numpy) style.

### CODE REVIEW

**Comment by StrikerRUS** [comment]
Is docstring OK? I tried to make it informative and simple at the same time, but I'm still not satisfied 😕

**Comment by guolinke** [comment]
@StrikerRUS it looks good to me

**Comment by StrikerRUS** [comment]
OK, thanks! Can we merge then or second review is needed?

**Comment by guolinke** [comment]
@StrikerRUS I think it is ready for merge

**Comment by StrikerRUS** [comment]
@guolinke I've found that it crashes with **categorical feature**. What do you think we can do with it? Simply raise error for cat. feature or is there any smarter solution?

**Comment by guolinke** [comment]
@StrikerRUS 
can we use a string type directly for the x-axis in histogram for categorical features?

**Comment by StrikerRUS** [comment]
`numpy.histogram()` crashes with string type. Also, how to sort cat. string values, let say `['0', '3', '2', '0||3']`?

**Comment by guolinke** [comment]
@StrikerRUS yeah, the combination is a problem. Let us skip the categorical features for now.

**Comment by StrikerRUS** [comment]
@guolinke OK. Created #2045 for this.
